### PR TITLE
Refactor FolderTree component

### PR DESCRIPTION
**Pull request by cmsetzer**
This branch reimplements the FolderTree component in the Vue dashboard. This component is used to display and interact with folders of Perma links. It drops the jQuery plugin [jstree](https://www.jstree.com/) in favor of [Headless Tree](https://headless-tree.lukasbach.com/), which is widely used in React and can be integrated with other frameworks.

Note: This PR's scope has expanded. Previously, the goal was to remove any extraneous jQuery from the tree component, but that turned out not to be worth the effort. Instead, I have tried to reimplement the tree entirely while also retaining jstree's accessibility support.

**Before:**

* jQuery dependency via jstree
* Two Vue components (JSTree and FolderTree). In principle, this was meant to separate the jstree-based implementation from the business logic of folder interactions, but in practice decoupling these layers is difficult and perhaps not viable
* Support for keyboard navigation of folder tree
* Support for mouse-based (but not keyboard-based?) drag and drop
* Test coverage: 2 tests

**After:**

* No jQuery dependency
* Vue component (FolderTree) + a Vue [composable](https://vuejs.org/guide/reusability/composables) for the Headless Tree object
* Support for keyboard navigation of folder tree
* Support for both mouse- and keyboard-based drag and drop
* Test coverage: 12 tests cover the main functionality and edge cases (create/rename/delete, root and sponsored folder restrictions, `folder` URL param, keyboard navigation, drag and drop)

### CODE REVIEW

**Review by rebeccacremona — COMMENTED** [review]
Great idea for an incremental improvement. I want to take a look at this tomorrow with a fresher brain: I want to check out some things that caught my eye ("Is it fine that `triggerOnWindow("FolderTreeModule.selectionChange"` is no longer firing?"; "Is is okay that the `// special handling for Sponsored Links parent folder` is deleted?"; etc.).

**Comment by cmsetzer** [comment]
Thanks for flagging the sponsored link parent expansion issue — that's an oversight on my part. I also made a couple CSS-related fixes:

* Restored behavior wherein we append a special `dragging` class to the body during drag and drop
* There was a pre-existing CSS bug (unrelated to this PR) that rendered the text selection background as white when renaming a folder; it now uses the default selection color

File: perma_web/static/css/style-responsive.scss [hunk 1]
```diff
@@ -3961,8 +3961,9 @@ footer {
     width: 100% !important;
     z-index: 1000;
     background: white;
-    &:focus, &::selection {
+    &:focus {
       background: white;
+      color: $color-black;
```
Comment (rebeccacremona) [inline perma_web/static/css/style-responsive.scss]: Interesting: I see this difference. When renaming a folder, I now see the text highlighted with my OS's standard color for selected text. That seems fine/intuitive to me :-) I see that -I- added that CSS in December of 2016. I wonder who wanted the highlight removed 😆

File: perma_web/static/frontend/components/JSTree.vue [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 
-const emit = defineEmits(['nodeSelect', 'nodeUnselect', 'nodeExpand', 'nodeCollapse']);
+const emit = defineEmits(['nodeSelect', 'nodeUnselect', 'nodeExpand', 'nodeCollapse', 'moveLink']);
 
 const folderTreeRef = ref(null);
-
-let APIModule = require('../../js/helpers/api.module.js');
-let Helpers = require('../../js/helpers/general.helpers.js');
-
-let localStorageKey = Helpers.variables.localStorageKey;
```
Comment (rebeccacremona) [inline perma_web/static/frontend/components/JSTree.vue:15]: I believe this was the only use of `Helpers.variables.localStorageKey` in the codebase; should we make bold to delete its definition from the `Helpers` module?

File: perma_web/static/frontend/components/JSTree.vue [hunk 3]
```diff
@@ -247,72 +231,81 @@ function apiFoldersToJsTreeFolders(apiFolders) {
...
-          }
-          parentFolder.state.opened = true;
 
-          // find the subfolders and load them in:
-          let apiResponse = apiResponses[i][0];
-          let subfolders = apiResponse ? apiResponse.objects : null;  // if API response doesn't make sense, we'll just stop loading the tree here
-          if (subfolders && subfolders.length) {
-            parentFolder.children = apiFoldersToJsTreeFolders(subfolders);
+  // Fetch contents of all folders in the saved path.
+  // Temporarily limit response to 500; TODO: handle pagination
```
Comment (rebeccacremona) [inline perma_web/static/frontend/components/JSTree.vue]: I know this is just a rewording of an already-present comment, but it threw me: I was like "the contents? oh oh oh, we are looking for a list of its subfolders... not the links in it"

**Review by rebeccacremona — COMMENTED** [review]
We have been chatting about this PR in Slack; here are a couple additional very tiny remarks 🙂 Thanks for all the back and forth about this one!

**Review by durkinm2 — APPROVED** [review]
LGTM!

**Comment by cmsetzer** [comment]
@durkinm2 and @rebeccacremona, I've updated the [title and description](https://github.com/harvard-lil/perma/pull/3758#issue-4108850268) above to reflect the updated scope of this PR. Would you mind giving it a re-review?

**Comment by cmsetzer** [comment]
I'm going to close this; we can reopen/revisit down the road if needed.
